kvbc/src/migrations/block_merkle_latest_ver_cf_migration.cpp

```diff
@@ -106,42 +111,55 @@ void BlockMerkleLatestVerCfMigration::clearExistingLatestVerCf() {
 
 void BlockMerkleLatestVerCfMigration::iterateAndMigrate() {
   auto blockchain = Blockchain{db_};
-  auto key_hashes = std::vector<Hash>{};
-  auto values = std::vector<::rocksdb::PinnableSlice>{};
-  auto statuses = std::vector<::rocksdb::Status>{};
-  for (auto block_id = INITIAL_GENESIS_BLOCK_ID; block_id <= blockchain.getLastReachableBlockId(); ++block_id) {
-    const auto block = blockchain.getBlock(block_id);
-    if (!block.has_value()) {
-      throw std::runtime_error{"Failed to load block ID = " + std::to_string(block_id)};
-    }
-    auto it = block->data.categories_updates_info.find(kExecutionProvableCategory);
-    if (it != block->data.categories_updates_info.cend()) {
-      const auto block_merkle_output = std::get_if<BlockMerkleOutput>(&it->second);
-      ConcordAssertNE(block_merkle_output, nullptr);
-      key_hashes.clear();
-      for (const auto& [key, _] : block_merkle_output->keys) {
-        (void)_;
-        key_hashes.push_back(hash(key));
-      }
-      values.clear();
-      statuses.clear();
-      auto batch = NativeWriteBatch{db_};
-      auto key_it = block_merkle_output->keys.cbegin();
-      db_->multiGet(kTempCf, key_hashes, values, statuses);
-      ConcordAssertEQ(key_hashes.size(), values.size());
-      ConcordAssertEQ(key_hashes.size(), statuses.size());
-      for (auto i = 0ull; i < values.size(); ++i) {
-        const auto& value = values[i];
-        const auto& status = statuses[i];
-        if (status.ok()) {
-          batch.put(BLOCK_MERKLE_LATEST_KEY_VERSION_CF, key_it->first, value);
-        } else {
-          // We expect that no pruning has occurred. Therefore, the latest version of the key cannot be missing.
-          throw std::runtime_error{"multiGet() failed, reason: " + status.ToString()};
+  ThreadPool tp(iteration_batch_size_);
+  for (auto block_id = INITIAL_GENESIS_BLOCK_ID; block_id <= blockchain.getLastReachableBlockId();
+       block_id += iteration_batch_size_) {
+    std::vector<std::future<void>> tasks;
+    std::vector<NativeWriteBatch> batches(iteration_batch_size_, NativeWriteBatch(db_));
```

Comment:
Given NativeWriteBatch takes the db_ as its state, may it suggest that NativeWriteBatch is not thread safe?

Reply:
thanks, @cloudnoize, for the review.
It may be possible that NativeWriteBatch is not threadsafe. So here we are creating a separate batch for each thread and then using that batch to hold all the keys read from DB using multiget. Each thread is writing in its own allocated batch as the batches vector is preallocated.